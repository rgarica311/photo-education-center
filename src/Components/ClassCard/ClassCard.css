

.frameWorkCard {
  height: 350px;
  width: 250px;
  position: relative;
  margin-bottom: 25px;
  margin-right: 15px;
}

.cardTop {
  height: 40%;
  width: 100%;
  background-color: slategrey;
  display: flex;
  justify-content: center;
  padding-top: 10px;
  font-family: 'Raleway', sans-serif;
  font-weight: 900;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  font-size: 1rem;
  color: white;
}

.cardImg {
  height: 100px;
  width: 100px;
  border-radius: 50%;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  position: absolute;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  top: 25%;
  border: 2px solid black;
  background-image : url('http://creativephotoacademy.com/wp-content/uploads/2016/01/classes-162x162.jpg');
}

.cardBottom {
  width: 100%;
  height: 50%;
  display: flex;
  align-items: flex-end;
  padding: 10px;
  line-height: 1.4em;
  padding-bottom: 25px;
  background-color: coral;
  font-size: .9rem;
  font-family: 'Work+Sans', sans-serif;
}

.download {
  position: absolute;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  background-color: red;
  width: 100%;
  height: 10%;
  border: none;
  font-family: 'Raleway', sans-serif;
  font-weight: bold;
  text-transform: uppercase;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
}

.download:hover {
  cursor: pointer;
  background-color: #0D324D;
  color: white;
}

.storyH1 {
  margin: 0 auto;
  width: 100%;
  height: 50px;
  position: absolute;
  background-color: #BFC5BD;
  text-align: center;
  top: 150px;
  z-index: 1;
  font-family: 'Raleway', sans-serif;
  font-weight: 300;
}

@media screen and (max-width: 749px){
  .storyH1 {
    z-index: 0;
  }


  .frameWorkCard {
    margin-right: 0;
  }

  .classesContainer {
    justify-content: center;
  }
}
