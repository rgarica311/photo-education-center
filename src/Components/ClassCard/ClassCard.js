import React, { Component } from 'react'
import './ClassCard.css'

export default class ClassCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      history: this.props.history
    }
  }


  async componentDidMount() {
    try {
      let response = await fetch('https://blooming-garden-38714.herokuapp.com/frameworks')
      if(!response.ok){
        throw new Error(response.status)
      } else {
        this.setState({
          frameworks: await response.json()

        })
      }
    } catch(error) {
      console.error('error', error)
    }


  }

  handleClick(path){
    console.log('path', path)
    this.state.history.push(`${path}`)
  }

  render(){
    return (
        <div className='frameWorkCard'>
          <div className='cardTop'>
            {this.props.title}
          </div>
          <div className='cardImg'></div>
          <div className='cardBottom'>
            {this.props.desc}
          </div>
          <button id='hero' onClick={e => this.handleClick(this.props.path)} className='download'>Click for Details</button>
      </div>
      )
  }
}
