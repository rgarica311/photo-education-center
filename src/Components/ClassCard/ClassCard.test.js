import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ClassCard from './ClassCard';

describe(`ClassCard Component`, () => {

  it('renders ClassCard by default', () => {
    const wrapper = shallow(<ClassCard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
