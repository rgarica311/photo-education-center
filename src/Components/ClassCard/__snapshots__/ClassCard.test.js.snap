// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassCard Component renders ClassCard by default 1`] = `
<div
  className="frameWorkCard"
>
  <div
    className="cardTop"
  />
  <div
    className="cardImg"
  />
  <div
    className="cardBottom"
  />
  <button
    className="download"
    id="hero"
    onClick={[Function]}
  >
    Click for Details
  </button>
</div>
`;
